Samples/auth/Outlook-Add-in-SSO-NAA-Identity/README.md

```diff
@@ -0,0 +1,249 @@
+---
+page_type: sample
+urlFragment: outlook-add-in-sso-naa-identity
+products:
+  - office
+  - office-outlook
+languages:
+  - javascript
+extensions:
+  contentType: samples
+  technologies:
+    - Add-ins
+  createdDate: "01/14/2025 10:00:00 AM"
+description: "This sample shows how to send identity claims across the network to resources like a web API service."
+---
+
+# Send identity claims to resources using nested app authentication (NAA) and SSO
+
+The scenario for this sample is that you need to send the signed-in user's identity claims (such as name, email, or a unique ID) to a resource such as a database. This enables you to associate and track data for each user in the resource. This sample replaces an obsolete pattern for legacy Exchange Online tokens. The obsolete pattern is that Exchange Online identity tokens were passed over the network to share user identity. Using Entra ID tokens, the ID token must never be passed by your Outlook add-in over the network to any resource. It can only be used within the context of the add-in (such as the task pane) itself. To pass user identity over the network, you use the Entra ID access token which contains the identity claims.
+
+This sample shows how to implement a todo list for multiple users with an Outlook add-in.
+
+1. The Outlook add-in (SPA client) uses the MSAL.js library to obtain an access token from Microsoft Entra ID. The MSAL.js library returns an id token, access token, and refresh token.
+1. The access token is passed in a web request to a protected web API service. The web request validates the access token and confirms the user has permissions to perform the action in the API. The access token contains any identity claims the web API needs.
+
+![A diagram showing the Outlook add-in getting an access token from Microsoft Entra ID and calling the protected web API with the access token.](architecture-diagram.png)
+
+
+
+## Applies to
+
+- Outlook on Windows (new and classic), Mac, mobile, and on the web.
+
+## Prerequisites
+
+- Office connected to a Microsoft 365 subscription (including Office on the web).
+- [Node.js](https://nodejs.org/) version 16 or greater.
+- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) version 8 or greater.
+
+## Register the sample applications with your Microsoft Entra tenant
```

Comment:
For each of these heading 3 level procedures, consider adding a screen shot.

Reply:
done